rfc-008-server-naming-convention.md

```diff
@@ -0,0 +1,55 @@
+# Server naming convention
+
+## Summary
+
+Have servers named in a consistent, descriptive and therefore helpful-at-a-glance way.
+
+## Problem
+
+We have no consistency when it comes to naming servers as we have no agreed convention to follow. This makes it difficult to quickly identify details about a server and resolve issues.
+
+## Proposal
+
+At a glance from the server name we would like to be able to identify:
+
+ 1. Customer
+ 2. Hosting provider
+ 3. Server location
+ 4. Environment
+ 5. Optional: Blue/Green if used
+ 6. Count/index of server
+ 7. Application/role
+
+### Proposed format
+
+```
+<app><count>-(blue|green)-<env>-<location>-<provider>-<customer>
```

Comment:
Based on thoughts about Endianness, and building on @zuren's "At-a-Glance" point, maybe something like:

```<customer>-<env>-<app><count>-(blue|green)-<provider>-<location>```

making `<customer>` optional if in their own Infra.

Reply:
Blue/Green probably shouldn't be optional. Ideally staging and production should be Blue/Green.

Reply:
What if you don't need blue/green cos you have rolling deploys on kubernetes?

Reply:
blue/green could also be used for previewing features in production.

Reply:
Yeah you could configure it to roll out a feature branch for temporary preview

Reply:
@lukemorton continuous environments (where used) don't _normally_ have blue/green setup. So that's why I was thinking have it optional. And for the reasons Craig mentioned.